app/controllers/katello/api/v2/content_view_filter_bulk_rules_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Katello
+  class Api::V2::ContentViewFilterBulkRulesController < Api::V2::ApiController
+    before_action :find_filter
+    before_action :find_rule, :except => [:create]
+
+    api :POST, "/content_view_filters/:content_view_filter_id/bulk",
+        N_("Create a bunch of filter rules. The parameters included should be based upon the filter type.")
+    param :content_view_filter_id, :number, :desc => N_("filter identifier"), :required => true
+    param :data, Array, :of => [Hash], :desc => N_("hash for RPM bulkloads: \
+                                                    [{'name': 'name', \
+                                                    'version': 'version',\
+                                                    'architecture': 'architecture'}]") do
+      param :name, String, :desc => N_("package: name"), :required => true
+      param :version, String, :desc => N_("package: version"), :required => false
+      param :architecture, String, :desc => N_("package: architecture"), :required => false
+    end
+    def create
+      rule_clazz = ContentViewFilter.rule_class_for(@filter)
+      if params[:data].is_a?(Array)
+        rules = params[:data].select{ |entry| entry[:name] and entry[:name] != ""}.map do |bulk_rule|
+          rule_clazz.create!([:filter => @filter, name: bulk_rule[:name],
+                                                  version: bulk_rule[:version],
+                                                  architecture: bulk_rule[:architecture]])
+        end
+      else
+        fail HttpErrors::BadRequest, _("Parameter 'data' must be an array of hash: " \
+                                       + "[{'name1', 'ver1', 'arch1'}, {'name2', 'ver2', 'arch2'}, ...]")
+      end
+
+      rules = rules.flatten
+      if rules.many?
+        if rules.count != params[:data].count
+          text = params[:data].select{ |entry| not entry[:name] or entry[:name] == ""}
```

Comment:
If you do use the `all?` method above to validated, you can just do a `.map { |entry| entry[:name] }` here since it will be validated that the key exists.

Reply:
fixed